index.js

```diff
@@ -31,402 +39,627 @@ console.error = ui.debugLog;
+      const ticketAddr = await addr(ticketAddress_n)
+      const poolAddr = await addr(poolAddress_n)
+      const returnAddr = await addr(returnAddress_n)
+      const zeroAddr = trezorHelpers.zeroAddr()
+      // Create split transactions to fund tickets.
+      const outpoints = [{
+        addr: ticketAddr,
+        amt: ticketPortion
+      },{
+        addr: poolAddr,
+        amt: poolFee
+      }]
+      const inputTxHash = await sendToAddrs(outpoints)
+      const inp = await wallet.getTransaction(wsvc, inputTxHash)
+      const decodedInp = await wallet.decodeTransaction(decodeSvc, inp.getTransaction());
+      const refTxs = [decodedInp].map(trezorHelpers.walletTxToRefTx);
+      // The outputs seem to be randomized. Find the one we want to use by
+      // checking for the correct amount.
+      function findOut(amt) {
+        var outs = decodedInp.array[6]
+        var outputN = 0
+        for (var i = 0; i < outs.length; i++) {
+          const out = outs[i]
+          if (out[0] == amt) {
+            outputN = out[1]
+            break
+          }
         }
-        log("Listing devices");
-        devices.map((d, i) => {
-            const feat = d.device.features;
-            if (d.state === "connected") {
-                log("Device %d (%s): %s (%s)", i, d.state, feat.device_id, feat.label);
-            } else {
-                log("Device %d (%s)", i, d.state);
-            }
+        if (outputN == null) outputN = 0
+        return outputN
+      }
+      const poolInput = {
+        address_n: poolAddress_n,
+        prev_hash: inputTxHash,
+        prev_index: findOut(poolFee),
+        amount: poolFee.toString(),
+      }
+      const ticketInput = {
+        address_n: ticketAddress_n,
+        prev_hash: inputTxHash,
+        prev_index: findOut(ticketPortion),
+        amount: ticketPortion.toString(),
+      }
+      const poolOPreturn = trezorHelpers.sstxcommitment(poolFeeAddr, poolFee, false)
+      const ticketOPreturn = trezorHelpers.sstxcommitment(returnAddr, ticketPortion, false)
+      const sstxsubmission = {
+          address: multiSig,
+          script_type: 'SSTXSUBMISSIONSH',
+          amount: price.toString()
+      }
+      const poolsstxcommitment = {
+          script_type: 'PAYTOOPRETURN',
+          op_return_data: poolOPreturn,
+          amount: "0"
+      }
+      const poolsstxchange = {
+          address: zeroAddr,
+          script_type: 'SSTXCHANGE',
+          amount: "0"
+      }
+      const ticketsstxcommitment = {
+          script_type: 'PAYTOOPRETURN',
+          op_return_data: ticketOPreturn,
+          amount: "0"
+      }
+      const ticketsstxchange = {
+          address: zeroAddr,
+          script_type: 'SSTXCHANGE',
+          amount: "0"
+      }
+      const inputs = [poolInput, ticketInput]
+      const outputs = [sstxsubmission, poolsstxcommitment, poolsstxchange, ticketsstxcommitment, ticketsstxchange]
```

Comment:
You shouldn't be constructing the ticket manually. Use the [PurchaseTickets](https://github.com/decred/dcrwallet/blob/master/rpc/documentation/api.md#purchasetickets) rpc call, setting `dont_sign_tx` to `true`, so that the underlying dcrwallet generates the unsigned ticket tx according to the current consensus rules.

That call will return the split tx and the ticket tx, both of which need signing.

Reply:
Sorry I missed your comments. Will do this.